### DISCUSSION TITLE
Page Transitions Leading To `(intermediate value)() is undefined` Error

### DISCUSSION THREAD

**Discussion by fweth** [vikejs/vike#1484] [Discussion]
I asked this on [SO](https://stackoverflow.com/questions/77924727/page-transitions-in-vike-react) before, how do we get proper page transitions in Vike/React?  I tried to implement via [React Transition Group](https://reactcommunity.org/react-transition-group/) (this package is already 2 years old, I hope it still follows all the correct APIs) but got an error when transitioning from a dynamic route (`/star-wars`) to a static route:

```
Uncaught TypeError: (intermediate value)() is undefined
    Page +Page.jsx:6
    React 11
    workLoop scheduler.development.js:266
    flushWork scheduler.development.js:239
    performWorkUntilDeadline scheduler.development.js:533
    ...
```
    
I also tried it with my own code, using `React.cloneElement`, which is also what RTG uses under the hood.  Here is what I did to the `Content` function in the boilerplate (with RTG):

```jsx
function Content({ children, url }) {
  const divRef = useRef();
  return (
    <div id="page-container">
      <SwitchTransition>
        <CSSTransition key={url} nodeRef={divRef} classNames="fade" timeout={400}>
          <div
            id="page-content"
            style={{
              padding: 20,
              paddingBottom: 50,
              minHeight: "100vh",
            }}
            ref={divRef}
          >
            {children}
          </div>
        </CSSTransition>
      </SwitchTransition>
    </div>
  );
}
```

Any ideas?  The React docs say:
> Using cloneElement is uncommon and can lead to fragile code. [See common alternatives.](https://react.dev/reference/react/cloneElement#alternatives)

But I wouldn't know what else to use?

**Comment by brillout** [comment]
Why not https://vike.dev/onPageTransitionStart?

**Reply by fweth** [reply]
Because I want to set a duration for the transition, I didn't realize how to achieve this with `onPageTransitionStart`. Also, when we want overlays or some kind of transitions where both, the old and the new page are visible for the given duration, I think we need to use `React.cloneElement` and I think this is what causes the error, even when I would put the code in `onPageTransitionStart` but I can experiment with that!

Maybe, if anybody could just help me understand what the error means, or when it typically occurs (I have never seen it before), I could do some more debugging and experimentation!

**Reply by brillout** [reply]
Makes sense. As for the React error, I ain't familiar with that one so I can't help much. But definitely let me know if you believe there is something Vike can help you with.

Keep us updated 👀 It could be nice to propagate your findings to `vike-react`.
